R/methods.R

```diff
@@ -1,26 +1,34 @@
 #' @export
 format.lint <- function(x, ..., width = getOption("lintr.format_width")) {
-  if (requireNamespace("cli", quietly = TRUE)) {
-    color <- switch(x$type,
-      warning = cli::col_magenta,
-      error = cli::col_red,
-      style = cli::col_blue,
-      cli::style_bold
-    )
-    emph <- cli::style_bold
+  color <- switch(x$type,
+    warning = cli::col_magenta,
+    error = cli::col_red,
+    style = cli::col_blue,
+    cli::style_bold
+  )
+  emph <- cli::style_bold
+
+  has_hyperlink_support <- cli::ansi_has_hyperlink_support()
+  if (has_hyperlink_support) {
+    fmt_start <- "{.path "
+    fmt_end <- "}"
+    fmt_f <- cli::format_inline
   } else {
-    # nocov start
-    color <- identity
-    emph <- identity
-    # nocov end
+    fmt_start <- ""
+    fmt_end <- ""
+    fmt_f <- paste0
   }
 
   annotated_msg <- paste0(
     emph(
-      x$filename, ":",
-      as.character(x$line_number), ":",
-      as.character(x$column_number), ": ",
-      sep = ""
+      fmt_f(
```

Comment:
I think this puts too much logic inside `format.lint`, could you add a helper for the hyperlink instead? e.g. (pseudo-code):

```r
styled_line_ref <- build_line_ref(x)
styled_type <- color(...)
annotated_msg <- glue("{line_ref}: {styled_type} [{linter}] {emph(message)}")
```

Reply:
Thanks for the suggestion, I also thought something felt off. I think it looks less hackish now?